src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -139,6 +139,29 @@
          [Spinner {:text (:banner props)}]]]))})
 
 
+(react/defc SafeBlocker
+  {:show
+   (fn [{:keys [props state]}]
+     (swap! state assoc :show-requested? true)
+     (js/setTimeout #(when (:show-requested? @state)
+                      (swap! state assoc :showing? true))
+                    (:delay-time props)))
+   :hide
+   (fn [{:keys [state]}]
+     (swap! state dissoc :showing? :show-requested?))
+   :get-default-props
+   (fn []
+     {:delay-time 200})
```

Comment:
Since the units (milliseconds) are not obvious, suggest the key be `:delay-time-ms`.